tsdb/db.go

```diff
@@ -1122,6 +1124,18 @@ func BeyondSizeRetention(db *DB, blocks []*Block) (deletable map[ulid.ULID]struc
 	return deletable
 }
 
+func getDirsSizeMatchingRegex(re *regexp.Regexp, dir string) (sizes int64, err error) {
```

Comment:
Let's keep it simpler, `GetCheckpointSize`, which only take the directory and part of `tsdb/checkpoint.go`. Plus, let there be only 1 way of detecting checkpoints. So you can re-use the `listCheckpoints` method in `checkpoint.go` and maybe break it down into 2 methods `listCheckpointsDir` and use it inside `listCheckpoints` (as you need directories here).